web-extensions/chrome/engine/datamodel.js

```diff
@@ -435,17 +435,27 @@ BmcDataAPI.prototype.updateComic = function(comicId, chapter, page, cb) {
          * once. We may actually want to accept that, though.
          * Similarly, we may want to allow backwards-tracking too.. ?
          */
-        if (payload.tracking.chapter > chapter
-            || (payload.tracking.chapter == chapter
-                && (payload.tracking.page && page
-                    && payload.tracking.page > page))) {
-            return cb(new Error(LOCALIZATION.getString('S22')));
+        if (payload.tracking.chapter !== undefined && payload.tracking.page !== undefined) {
+            const trackChapter = parseInt(payload.tracking.chapter);
+            const trackPage = parseInt(payload.tracking.page);
+            const currentChapter = parseInt(chapter);
+            const currentPage = parseInt(page);
+
+            if (trackChapter > currentChapter || trackPage > currentPage) {
```

Comment:
nope,  trackPage might be higher than currentPage, while currentChapter is higher than trackChapter.
We need to check that either:
1. trackChapter is strictly greater than currentChapter
2. trackChapter is equal to currentChapter AND trackPage is greater than currentPage

Here, we're failing the second check, and possibly creating another bug :)

Reply:
Good catch!